modules/core/src/smithy4s/http/HttpUnaryServerRouter.scala

```diff
@@ -31,6 +31,28 @@ object HttpUnaryServerRouter {
     */
   def apply[Alg[_[_, _, _, _, _]], F[_], Request, Response](
       service: smithy4s.Service[Alg]
+  )(
+      impl: service.Impl[F],
+      makeServerCodecs: UnaryServerCodecs.Make[F, Request, Response],
+      endpointMiddleware: Endpoint.Middleware[Request => F[Response]],
+      getMethod: Request => HttpMethod,
+      getUri: Request => HttpUri,
+      addDecodedPathParams: (Request, PathParams) => Request
+  )(implicit F: MonadThrowLike[F]): Request => Option[F[Response]] = {
+    new KleisliRouter[Alg, service.Operation, F, Request, Response](
+      service,
+      service.toPolyFunction[smithy4s.kinds.Kind1[F]#toKind5](impl),
+      makeServerCodecs,
+      endpointMiddleware,
+      getMethod,
+      getUri,
+      addDecodedPathParams,
+      encodeErrorsBeforeMiddleware = false
+    )
+  }
+
+  def v2[Alg[_[_, _, _, _, _]], F[_], Request, Response](
```

Comment:
I’m not a fan of this approach either, but I wasn’t able to overload `apply` above - likely due to the second argument list

If you know a better way to maintain binary compatibility or can suggest a more suitable method name, I’d really appreciate it

Reply:
The only reason for the currying is the path-dependant `Impl` type (on the `impl` parameter). Therefore, you could simply add the boolean parameter on the first parameter set, which would allow to overload to `apply`